src/main/scala/org/clulab/asist/agents/DacClient.scala

```diff
@@ -50,82 +45,108 @@ class DacClient (
   ): Unit = {
     val message = read[DialogAgentMessage](json)
     val data = message.data
-    val requestJson = write(new DialogActClassifierMessage(
-      data.participant_id,
-      data.text,
-      data.extractions)
+    val requestJson = write(
+      new DialogActClassifierMessage(
+        Option(data.participant_id).getOrElse(""),
+        Option(data.text).getOrElse(""),
+        Option(data.extractions).getOrElse(Seq())
+      )
     )
     val request = HttpRequest(
-      uri = Uri("http://localhost:8000/classify"),
+      uri = Uri(s"${serverLocation}/classify"),
       entity = HttpEntity(ContentTypes.`application/json`,requestJson)
     )
     val future: Future[HttpResponse] = Http().singleRequest(request)
-    val response: HttpResponse = Await.ready(future, 10 seconds).value.get.get
-    val futureClassification: Future[Classification] = response.entity.dataBytes
-      .runReduce(_ ++ _)
-      .map{ line =>
-        val ret = line.utf8String.split(" ").headOption.map(Classification)
-        ret.getOrElse(new Classification(""))
-      }
 
-    futureClassification onComplete {
-      case Success(c: Classification) =>  
-        agent.parseJValue(json).toList.map{metadata =>
+    try {
+      val response: HttpResponse = Await.ready(future, 10 seconds).value.get.get
+      val futureClassification: Future[Classification] = response.entity.dataBytes
+        .runReduce(_ ++ _)
+        .map{ line =>
+          val ret = line.utf8String.split(" ").headOption.map(Classification)
+          ret.getOrElse(new Classification(""))
+        }
+
+      futureClassification onComplete {
+        case Success(c: Classification) =>  
+          showStatus(message.header.timestamp, response.status)
           val label = c.name.replace("\"","")
           val newData = data.copy(dialog_act_label = label)
           val newMetadata = metadata.replace(
             "data"::Nil,
             Extraction.decompose(newData)
           )
           val newMetadataJson = write(newMetadata)
-          finishClassification(agent.addDacQuery(s), newMetadataJson)
-        }
-      case Failure(t) =>
-        logger.error(s"An error occured:  ${t}")
-        finishClassification(agent.addError(s), json)
+          val queried = s.copy(dacQueries = s.dacQueries + 1)
+          val done = agent.writeLine(queried, newMetadataJson)
+          agent.iteration(done)
+        case Failure(t) =>
+          showStatus(message.header.timestamp, response.status)
+          logger.error(s"An error occured:  ${t}")
+          val done = s.copy(errors = s.errors + 1)
+          agent.iteration(done)
+      }
+    } catch {
+      case NonFatal(t) => 
+        logger.error("Error processing {}", json)
+        logger.error("{}",t.toString)
+        val done = s.copy(terminated = true)
+        agent.iteration(done)
     }
   }
 
+  def showStatus(annotation: String, sc: StatusCode): Unit = {
+    logger.info("{} HttpResponse status = {}", annotation, sc.value)
```

Comment:
same question as above

Reply:
Same answer as above.   If you think the '{}',value syntax isn't good style, I can change these.

Reply:
FWIW I feel pretty strongly that the string interpolation syntax  

```scala
logger.info(s"${annotation} HttpResponse status = ${sc.value}")
```

is more readable. Plus, I think it can be considered idiomatic, since it's a core language feature as of Scala 2.10.